### PR TITLE
[docs] Update /connector docs

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Update /connector docs for new public APIs

### CODE REVIEW

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 1]
```diff
@@ -470,7 +470,119 @@ If the `show_command` parameter is given, the API call will instead return the g
...
 Get the list of configured [connectors](/administrator/configuration/connectors/):
 
-    curl -L -X POST demo.gethue.com/desktop/connectors/api/instances
+    curl -X GET 'https://demo.gethue.com/api/connector/instances' -H 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjI2MjM3OTcxLCJqdGkiOiJlYjI1NDg5N2ZlNWI0NjZhODk3MDc5MWFmYjcxOWJiNyIsInVzZXJfaWQiOjExMDA3MTR9.9vfeC9dfmRoYM586GzwnlcCCWcqCciszAhoDFwsSIPs' -H 'Content-Type: application/x-www-form-urlencoded'
+
+### Types
+
+    curl -X GET 'https://demo.gethue.com/api/connector/types' -H 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjI2MjM3OTcxLCJqdGkiOiJlYjI1NDg5N2ZlNWI0NjZhODk3MDc5MWFmYjcxOWJiNyIsInVzZXJfaWQiOjExMDA3MTR9.9vfeC9dfmRoYM586GzwnlcCCWcqCciszAhoDFwsSIPs' -H 'Content-Type: application/x-www-form-urlencoded'
+    
+### New connector
```
Comment (romainr) [inline docs/docs-site/content/developer/api/rest/_index.md:479]: Skip 'connector' in all the subtitles are redundant

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 2]
```diff
@@ -470,7 +470,119 @@ If the `show_command` parameter is given, the API call will instead return the g
 
 Get the list of configured [connectors](/administrator/configuration/connectors/):
 
-    curl -L -X POST demo.gethue.com/desktop/connectors/api/instances
+    curl -X GET 'https://demo.gethue.com/api/connector/instances' -H 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjI2MjM3OTcxLCJqdGkiOiJlYjI1NDg5N2ZlNWI0NjZhODk3MDc5MWFmYjcxOWJiNyIsInVzZXJfaWQiOjExMDA3MTR9.9vfeC9dfmRoYM586GzwnlcCCWcqCciszAhoDFwsSIPs' -H 'Content-Type: application/x-www-form-urlencoded'
```
Comment (romainr) [inline docs/docs-site/content/developer/api/rest/_index.md:473]: Example of output?

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 3]
```diff
@@ -470,7 +470,119 @@ If the `show_command` parameter is given, the API call will instead return the g
...
+
+### Types
+
+    curl -X GET 'https://demo.gethue.com/api/connector/types' -H 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjI2MjM3OTcxLCJqdGkiOiJlYjI1NDg5N2ZlNWI0NjZhODk3MDc5MWFmYjcxOWJiNyIsInVzZXJfaWQiOjExMDA3MTR9.9vfeC9dfmRoYM586GzwnlcCCWcqCciszAhoDFwsSIPs' -H 'Content-Type: application/x-www-form-urlencoded'
```
Comment (romainr) [inline docs/docs-site/content/developer/api/rest/_index.md:477]: same

Comment (Harshg999) [reply docs/docs-site/content/developer/api/rest/_index.md:477]: It is a very long output which can clutter the docs.

Comment (Harshg999) [reply docs/docs-site/content/developer/api/rest/_index.md:477]: Shall we add some part of it just to give a glimpse of how it looks @romainr ?

Comment (romainr) [reply docs/docs-site/content/developer/api/rest/_index.md:479]: Putting just 1 item should fit

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 4]
```diff
@@ -470,7 +470,131 @@ If the `show_command` parameter is given, the API call will instead return the g
...
+    ...........
+    ] }, { "category": "optimizers", "category_name": "Optimizers", "description": "", "values": [ { "nice_name": "Optimizer", "dialect": "optimizer", "settings": [], "category": "optimizers", "description": "", "properties": {} } ] }, { "category": "schedulers", "category_name": "Schedulers", "description": "", "values": [ { "nice_name": "Oozie", "dialect": "oozie", "settings": [], "category": "schedulers", "description": "", "properties": {} }, 
+    ...........
+    ] }, { "category": "plugins", "category_name": "Plugins", "description": "", "values": [] } ], "categories": [ { "name": "Editor", "type": "editor", "description": "" }, { "name": "Browsers", "type": "browsers", "description": "" }, { "name": "Catalogs", "type": "catalogs", "description": "" }, { "name": "Optimizers", "type": "optimizers", "description": "" }, { "name": "Schedulers", "type": "schedulers", "description": "" }, { "name": "Plugins", "type": "plugins", "description": "" } ] }
+    
+### New
```
Comment (romainr) [inline docs/docs-site/content/developer/api/rest/_index.md:491]: Create

**Review by romainr — APPROVED** [review]

**Pull request merged**